gcloud/datastore/__init__.py

```diff
@@ -170,4 +182,7 @@ def allocate_ids(incomplete_key, num_ids):
     :rtype: list of :class:`gcloud.datastore.key.Key`
     :returns: The (complete) keys allocated with `incomplete_key` as root.
     """
-    return _require_dataset().allocate_ids(incomplete_key, num_ids)
+    dataset = _require_dataset()
+    connection = _require_connection()
+    return connection_module.allocate_ids(incomplete_key, num_ids,
```

Comment:
Indeed, but internals using protobufs is a bit different than what happens on `Connection`, which is inputs and outputs that are protobufs.